app/questionnaire/path_finder.py

```diff
@@ -109,11 +107,22 @@ def _build_path_within_group(self, blocks, block_index, this_location, path):
             if block_index is None:
                 return path, prev_block_index
 
+            block = blocks[block_index]['block']
+
+            if block.get('skip_conditions') and \
+                    evaluate_skip_conditions(block['skip_conditions'], self.schema, self.metadata,
+                                             self.answer_store, this_location.group_instance, routing_path=path):
+
+                if block_index < len(blocks) - 1:
+                    this_location = Location(blocks[block_index + 1]['group_id'],
+                                             blocks[block_index + 1]['group_instance'],
+                                             blocks[block_index + 1]['block']['id'])
+                    continue
+                return path, block_index
```

Comment:
I think this would be easier to understand if we returned before changing `this_location`, something like:

```python
if block_index == len(blocks) - 1:
	return path, block_index

this_location = Location(blocks[block_index + 1]['group_id'],
                         blocks[block_index + 1]['group_instance'],
                         blocks[block_index + 1]['block']['id'])
continue
```